---
title: "Pbmc_template"
author: "LB"
date: "2/6/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
#**Package instalation and setup**

```{r, echo=TRUE}
install.packages("Seurat", repos = "http://cran.us.r-project.org")
library(Seurat)
```
#**Read in data and create the Seurat object**
```{r, echo=TRUE}

d_1 <- Read10X("/Volumes/scratch/agutierrez/10x_fastq/move/BCTCSF_0001_3_CS_MNC_C2_X5SCR_F00452_HWJNCBBXX/outs/filtered_gene_bc_matrices/GRCh38")
colnames(d_1) = paste0("F00452", sep = "_", colnames(d_1))
pbmc <- CreateSeuratObject(d_1, project = "F00452", min.genes = 200)
rm(d_1)
```
#**Data processing**  
##Filter cells by nUMI and log normalize data
```{r, echo=TRUE}
pbmc <- FilterCells(pbmc, "nUMI", 1000, Inf)
pbmc <- NormalizeData(pbmc)
```
##Find variable genes and scale data
```{r, echo=TRUE}
pbmc <- FindVariableGenes(pbmc)
pbmc <- ScaleData(pbmc)
```
##Run PCA
```{r, echo=TRUE}
pbmc <- RunPCA(object = pbmc, pc.genes = pbmc@var.genes, do.print = TRUE, pcs.print = 1:5, genes.print = 5)
```  
##Find clusters

```{r, echo=TRUE}
pbmc <- FindClusters(object = pbmc, reduction.type = "pca", dims.use = 1:10, resolution = 0.6, print.output = 0, save.SNN = TRUE)
```  
#**Run tSNE plot**
```{r, echo=TRUE}
pbmc <- RunTSNE(object = pbmc, dims.use = 1:10, do.fast = TRUE)
TSNEPlot(pbmc)
```  
##Plot gene expression (MS4A1 is a B cell gene)
```{r, echo=TRUE}
FeaturePlot(pbmc, "MS4A1")
VlnPlot(pbmc,"MS4A1")
```  
##Annotation for the tSNE plot
```{r, echo=TRUE}
canonical_markers <- c("IL7R", "CD14", "LYZ", "MS4A1", "CD8A", "FCGR3A", "MS4A7", "GNLY", "NKG7", "FCER1A", "CST3", "PPBP")
DotPlot(pbmc, genes.plot = canonical_markers, x.lab.rot = TRUE, plot.legend = TRUE)

current.cluster.ids <- c(0, 1, 2, 3, 4, 5, 6, 7)
new.cluster.ids <- c("CD4 T cells", "CD14+ Monocytes", "B cells", "CD8 T cells", 
    "FCGR3A+ Monocytes", "NK cells", "Dendritic cells", "Megakaryocytes")
pbmc@ident <- plyr::mapvalues(x = pbmc@ident, from = current.cluster.ids, to = new.cluster.ids)
TSNEPlot(object = pbmc, do.label = TRUE, pt.size = 0.5)
```
